spec/generators/module_spec.rb

```diff
@@ -76,13 +76,25 @@
       include_context 'mock metadata.json'
 
       let(:temp_target_dir) { '/path/to/temp/dir' }
+      let(:target_parent_writeable) { true }
 
       before(:each) do
         allow(File).to receive(:exist?).with(target_dir).and_return(false)
         allow(PDK::Util).to receive(:make_tmpdir_name).with(anything).and_return(temp_target_dir)
         allow(FileUtils).to receive(:mv).with(temp_target_dir, target_dir)
         allow(PDK::Util::Version).to receive(:version_string).and_return('0.0.0')
         allow(described_class).to receive(:prepare_module_directory).with(temp_target_dir)
+        allow(File).to receive(:writable?).with(File.dirname(target_dir)).and_return(target_parent_writeable)
+      end
+
+      context 'when the parent directory of the target is not writable' do
+        let(:target_parent_writeable) { false }
+
+        it 'raises a FatalError' do
+          expect {
```

Comment:
The ways I tried doing this as a unit test were all pretty clunky a likely to just silently continue passing if the underlying code was refactored. I'm inclined to just merge as-is.